.php-cs-fixer.php

```diff
@@ -2,24 +2,16 @@
 
 $finder = PhpCsFixer\Finder::create()
     ->in(__DIR__)
-    ->notPath([
-        '.github',
-        'cypress',
-        'dtd',
-        'js',
-        'lib',
-        'locale',
-        'registry',
-        'schemas',
-        'styles',
-        'templates',
-        'xml',
-    ])
     ->name('*.php')
-    ->name('_ide_helper')
-    ->notName('*.blade.php')
+    // The next two rules are enabled by default, kept for clarity
     ->ignoreDotFiles(true)
-    ->ignoreVCS(true);
+    ->ignoreVCS(true)
+    // The pattern is matched against each found filename, thus:
+    // - The "/" is needed to avoid having "vendor" match "Newsvendor.php"
+    // - The presence of "node_modules" here doesn't prevent the Finder from recursing on it, so we merge these paths below at the "exclude()"
+    ->notPath($ignoredDirectories = ['cypress/', 'js/', 'locale/', 'node_modules/', 'styles/', 'templates/', 'vendor/'])
+    // Ignore root based directories
+    ->exclude(array_merge($ignoredDirectories, ['dtd', 'lib', 'registry', 'schemas', 'xml']));
```

Comment:
The `notPath()` is called while the recursion is in progress, for each matched file. Which means, there's no way to break the loop (e.g. skip the whole `node_modules`).